src/odemis/driver/xt_client.py

```diff
@@ -349,20 +468,579 @@ def set_autofocusing(self, channel, state):
+        pos (dict string -> float): the new position for a moveAbs()
+        raise ValueError: if the argument is incorrect
+        """
+        for axis, val in pos.items():
+            if axis in self.axes:
+                axis_def = self.axes[axis]
+                if hasattr(axis_def, "choices") and val not in axis_def.choices:
+                    raise ValueError("Unsupported position %s for axis %s"
+                                     % (val, axis))
+                elif (hasattr(axis_def, "range") and not
+                      axis_def.range[0] <= val <= axis_def.range[1]):
+                    # TODO: if not referenced, double the range
+                    rng = axis_def.range
+                    raise ValueError("Position %s for axis %s outside of range %f->%f"
+                                     % (val, axis, rng[0], rng[1]))
+            else:
+                raise ValueError("Unknown axis %s" % (axis,))
+
+    def stop(self, axes=None):
+        """Stop the movement of the stage."""
+        self._executor.cancel()
+        self.parent.stop_stage_movement()
+        try:
+            self._updatePosition()
+        except Exception:
+            logging.exception("Unexpected failure when updating position")
+
+    def _createFuture(self):
+        """
+        Return (CancellableFuture): a future that can be used to manage a move
+        """
+        f = CancellableFuture()
+        f._moving_lock = threading.Lock()  # taken while moving
+        f._must_stop = threading.Event()  # cancel of the current future requested
+        f._was_stopped = False  # if cancel was successful
+        f.task_canceller = self._cancelCurrentMove
+        return f
+
+    def _cancelCurrentMove(self, future):
+        """
+        Cancels the current move (both absolute or relative). Non-blocking.
+        future (Future): the future to stop. Unused, only one future must be
+         running at a time.
+        return (bool): True if it successfully cancelled (stopped) the move.
+        """
+        # The difficulty is to synchronise correctly when:
+        #  * the task is just starting (not finished requesting axes to move)
+        #  * the task is finishing (about to say that it finished successfully)
+        logging.debug("Cancelling current move")
+        future._must_stop.set()  # tell the thread taking care of the move it's over
+        self.parent.stop_stage_movement()
+
+        with future._moving_lock:
+            if not future._was_stopped:
+                logging.debug("Cancelling failed")
+            return future._was_stopped
+
+
+class Focus(model.Actuator):
+    """
+    This is an extension of the model.Actuator class. It provides functions for
+    moving the SEM focus (as it's considered an axis in Odemis)
+    """
+
+    def __init__(self, name, role, parent, **kwargs):
+        """
+        axes (set of string): names of the axes
+        """
+
+        self.parent = parent
```

Comment:
You don't need to store it explicitly, as the super class will take care of it. Just use `parent` instead of `self.parent` in the line below.